pandas/io/pytables.py

```diff
@@ -313,10 +313,14 @@ def read_hdf(path_or_buf, key=None, **kwargs):
 
         """
 
-    if kwargs.get('mode', 'a') not in ['r', 'r+', 'a']:
+    # Ignore any supplied mode, we only need to read data
```

Comment:
I am not averse to this change, but its different. There is an actual issue here where a lower level function is mistakenly passed the mode arg. Pls solve that issue first.